### PR TITLE
Alluxio-UI fix bugs

### PR DESCRIPTION
**Pull request by WilliamZapata**
Fixes the following bugs
- UI
  - Pagination control should calculate the current page, next and previous page correctly, also change pagination icons.
  - Pages with links will no longer carry over the current queryString when clicked.
  - Fix the interpretation of the file pinned value received from the server.
  - Simplify the launch script for the UI (now we use npm for all ways, no need to know lerna exists).
  - Prevent forms from automatically submitting when the enter key is pressed on an input field.
  - Ensure debug fields appear for the browse view, not the logs view (oops).
  - Simplify theming by using a packaged import from bootswatch.
  - Use monospaced font for file names and file permissions.
- API
  - UIFileInfo
    - Return correct creation and modification dates as part of this class.
    - Return a map of percentages used by tier as part of the fileInfo object.
    - Pass serverConfiguration object as needed to allow the above.
  - UIBlockInfo
    - Pass serverConfiguration object to be able to calculate percentage in highest tier.
  - AlluxioMasterRestServiceHandler and AlluxioWorkerRestServiceHandler
    - Refactor code to pass serverConfiguration to the above classes.
    - Prevent 500 errors associated with attempting to retrieve data that is out of bounds with large enough values of offset and limit.

Screenshots:
screen shot 2019-02-15 at 3 23 35 pm
screen shot 2019-02-15 at 3 23 38 pm
screen shot 2019-02-15 at 3 23 41 pm
screen shot 2019-02-15 at 3 23 43 pm
screen shot 2019-02-15 at 3 23 45 pm
screen shot 2019-02-15 at 3 23 48 pm
screen shot 2019-02-15 at 3 23 50 pm
screen shot 2019-02-15 at 3 24 02 pm
screen shot 2019-02-15 at 3 24 05 pm
screen shot 2019-02-15 at 3 24 07 pm
screen shot 2019-02-15 at 3 24 09 pm

### CODE REVIEW

**Comment by calvinjia** [comment]
@WilliamZapata Could you provide a summary of the bugs fixed in this PR?

**Comment by apc999** [comment]
@WilliamZapata I see in RC-1, the "Lost Workers" is still shown as "Alluxio Configuration". do you fix this issue in this PR or it is already fixed after RC-1 ?

**Comment by WilliamZapata** [comment]
Oh I think that is part of a different pr that has not yet merged. I can cherry pick that.
-------- Original message --------From: Bin Fan <notifications@github.com> Date: 2/14/19  9:56 AM  (GMT-08:00) To: Alluxio/alluxio <alluxio@noreply.github.com> Cc: William Zapata <william@alluxio.com>, Mention <mention@noreply.github.com> Subject: Re: [Alluxio/alluxio] Alluxio-UI fix bugs (#8396) @WilliamZapata I see in RC-1, the "Lost Workers" is still shown as "Alluxio Configuration". do you fix this issue in this PR or it is already fixed after RC-1 ?

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Alluxio/alluxio","title":"Alluxio/alluxio","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/Alluxio/alluxio"}},"updates":{"snippets":[{"icon":"PERSON","message":"@apc999 in #8396: @WilliamZapata I see in RC-1, the \"Lost Workers\" is still shown as \"Alluxio Configuration\". do you fix this issue in this PR or it is already fixed after RC-1 ?"}],"action":{"name":"View Pull Request","url":"https://github.com/Alluxio/alluxio/pull/8396#issuecomment-463726225"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/Alluxio/alluxio/pull/8396#issuecomment-463726225",
"url": "https://github.com/Alluxio/alluxio/pull/8396#issuecomment-463726225",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by apc999** [comment]
Leave this to you and Calvin whether to cherry-pick. But just to make sure we fix it in RC-2

File: alluxio-ui/master/src/containers/pages/Browse/Browse.tsx [hunk 1]
```diff
@@ -195,40 +196,74 @@ export class Browse extends React.Component<AllProps, IBrowseState> {
...
-            <Button tag={Link} to={`/browse?path=${path}${queryStringSuffix}`} color="primary"
+            <Button tag={Link} to={`/browse?path=${path}${queryStringSuffix}`} color="secondary"
                     disabled={path === lastFetched.path}>Go</Button>
           </FormGroup>
         </Form>
         <Table hover={true}>
           <thead>
           <tr>
-            <th/>
+            <th></th>
```
Comment (calvinjia) [inline alluxio-ui/master/src/containers/pages/Browse/Browse.tsx]: Could you comment that this is to leave an icon placeholder? Also is the shorthand no longer preferred?

Comment (WilliamZapata) [reply alluxio-ui/master/src/containers/pages/Browse/Browse.tsx]: Great catch, the linter settings changed when we removed react-scripts-ts, I think about adding explicit checking for empty tags soon.

**Comment by WilliamZapata** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by WilliamZapata** [comment]
Jenkins, test this please.

**Pull request merged**